activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -141,6 +141,10 @@ def supports_datetime_with_precision?
         end
       end
 
+      def supports_virtual_columns?
+        !mariadb? && version >= "5.7.5"
```

Comment:
`supports_virtual_columns?` is only used in schema **dumping**.
https://github.com/rails/rails/pull/22589/files#diff-621ba9ae39e5952c32d9f06ecf8eb433R63
https://github.com/rails/rails/pull/22589/files#diff-f50ea2a3acc95dd2652e9cb1b3f75961R18

It is because schema creation support is easy, but schema dumping support is not easy. Current implementation (extracting `as` expression) is depend on `information_schema`, but I couldn't find the information in MariaDB at least 10.1.

So I dicided to only support schema creation for MariaDB (does not support schema dumping yet now).